src/plugins/intel_cpu/src/nodes/reverse_sequence.cpp

```diff
@@ -99,88 +80,109 @@ void ReverseSequence::initSupportedPrimitiveDescriptors() {
                          impl_desc_type::ref_any);
 }
 
-void ReverseSequence::execute(dnnl::stream strm) {
-    size_t i;
-    const float *src_data = reinterpret_cast<const float *>(getParentEdgeAt(REVERSESEQUENCE_DATA)->getMemoryPtr()->GetPtr());
-    float* dst_data = reinterpret_cast<float *>(getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetPtr());
-
-    switch (getParentEdgeAt(REVERSESEQUENCE_LENGTHS)->getMemory().getDesc().getPrecision()) {
-        case Precision::FP32: {
-            float *seq_lengths_data = reinterpret_cast<float *>(getParentEdgeAt(REVERSESEQUENCE_LENGTHS)->getMemoryPtr()->GetPtr());
-            for (i = 0; i < src_dims[batch_axis]; i++) {
-                if (static_cast<int32_t>(seq_lengths_data[i]) > static_cast<int>(src_dims[seq_axis])) {
-                    std::string errorMsg = "Incorrect input 'seq_lengths' values!";
-                    IE_THROW() << errorMsg;
-                }
-            }
+void ReverseSequence::prepareParams() {
+    const auto& dataMemPtr = getParentEdgeAt(REVERSESEQUENCE_DATA)->getMemoryPtr();
+    const auto& seqLengthsMemPtr = getParentEdgeAt(REVERSESEQUENCE_LENGTHS)->getMemoryPtr();
+    const auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+
+    if (!dataMemPtr || !dataMemPtr->isAllocated())
+        IE_THROW() << errorPrefix << " has not allocated input memory of 'data'";
+    if (!seqLengthsMemPtr || !seqLengthsMemPtr->isAllocated())
+        IE_THROW() << errorPrefix << " has not allocated input memory of 'seq_lengths'";
+    if (!dstMemPtr || !dstMemPtr->isAllocated())
+        IE_THROW() << errorPrefix << " has not allocated output memory";
+    if (getSelectedPrimitiveDescriptor() == nullptr)
```

Comment:
I think it would be better to place this check in the very beginning of the method - to avoid useless execution of lines 84-93 when this condition is met.

Reply:
But there's no guarantee of the execution sequence, compiler may reorder statements in this particular part as it likes.